src/net/sourceforge/kolmafia/textui/parsetree/Operation.java

```diff
@@ -3,11 +3,16 @@
 import java.io.PrintStream;
 import net.sourceforge.kolmafia.textui.AshRuntime;
 import net.sourceforge.kolmafia.textui.DataTypes;
+import net.sourceforge.kolmafia.textui.Parser;
 
 public class Operation extends Expression {
   private final Operator oper;
 
   public Operation(final Evaluable lhs, final Evaluable rhs, final Operator oper) {
+    super(
+        rhs == null
+            ? Parser.mergeLocations(oper.getLocation(), lhs.getLocation())
+            : Parser.mergeLocations(lhs.getLocation(), rhs.getLocation()));
```

Comment:
Assuming `mergeLocations` is commutable, perhaps

```suggestion
    super(Parser.mergeLocations(lhs.getLocation(), rhs == null ? oper.getLocation() : rhs.getLocation());
```
would be better.

Perhaps nicer would be an overload for `mergeLocations` that takes objects that extend `Command`, and gets the locations itself

Reply:
1- No, `mergeLocations` is not commutable. The first argument is the left-most bound, and the second argument is the right-most.

Wouldn't that mean a lot of type checking and casting (in the overloaded method, I mean)? I don't see how that would end up simpler...

Reply:
1. Makes sense!
2. No - if the compiler knows that an object extends `Command` then you can just call `.getLocation()` on it!

Reply:
Oh, I see, I thought you meant a different handling for the various...
Yeah, my bad. That seems good.

Reply:
@fredg1 Wait, that makes `mergeLocations` a little easily misused then - shouldn't it just take `start = min(a.start, b.start)` and `end = max(a.end, b.end)`?

Reply:
Not recommended.

Later on, we'll make it so that the parsing continues even when we see errors.
Even then, we'll want to calculate locations.
We rely on this behaviour when making them, because we don't want a location to shrink because one of its component was smaller than expected due to a parsing error.